File: staff-ui/src/types/role.ts
// Role and permission types matching backend DTOs

export interface RolePermissions {
  // Scheduling permissions
  schedulingView: boolean;
  schedulingManage: boolean;

  // Time-off permissions
  timeOffView: boolean;
  timeOffManage: boolean;
  timeOffApprove: boolean;

  // Staff permissions
  staffView: boolean;
  staffManage: boolean;

  // Service permissions
  servicesView: boolean;
  servicesManage: boolean;

  // Client permissions
  clientsView: boolean;
  clientsManage: boolean;

  // Report permissions
  reportsView: boolean;
  reportsExport: boolean;

  // Settings permissions
  settingsView: boolean;
  settingsManage: boolean;
}

export interface RoleResponse {
  id: string;
  businessId: string;
  name: string;
  description: string | null;
  isDefault: boolean;
  isImmutable: boolean;
  permissions: RolePermissions;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRoleRequest {
  name: string;
  description?: string;
  permissions?: RolePermissions;
  cloneFromRoleId?: string;
}

export interface UpdateRoleRequest {
  name?: string;
  description?: string;
  permissions?: RolePermissions;
}

export interface PermissionAction {
  name: string;
  description: string;
  enabled: boolean;
}

export interface PermissionArea {
  area: string;
  actions: PermissionAction[];
}

export interface AssignRoleRequest {
  staffMemberId: string;
  locationId: string;
  roleId: string;
}

File: staff-api/staff-domain/Entities/Role.cs
using System.ComponentModel.DataAnnotations;
using staff_domain.ValueObjects;

namespace staff_domain.Entities;

public class Role
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid BusinessId { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    public bool IsDefault { get; set; } = false;

    public bool IsImmutable { get; set; } = false;

    public RolePermissions Permissions { get; set; } = new RolePermissions();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
